### ISSUE TITLE
[RLlib] `hidden_layer_use_layernorm` doesn't get applied when set.

### ISSUE THREAD

**Issue by MatthewCWeston** [ray-project/ray#60926]
### What happened + What you expected to happen

The `hidden_layer_use_layernorm` attribute should apply a LayerNorm layer to the model architecture when set, but it does not. This is because catalog.py fails to apply the setting when instantiating an `MLPEncoderConfig` on [line 292](https://github.com/ray-project/ray/blob/master/rllib/core/models/catalog.py#L292). The fix I'm presently using is below:

```
...
encoder_config = MLPEncoderConfig(
                    input_dims=observation_space.shape,
                    hidden_layer_dims=hidden_layer_dims,
                    hidden_layer_activation=activation,
                    hidden_layer_use_layernorm=model_config_dict[
                        "fcnet_use_layernorm"
                    ],
                    ...
```

I should note that `fcnet_use_layernorm` should also be added to [default_model_config.py](https://github.com/ray-project/ray/blob/4ab83f0d598cd44e43c9eb2225ee49efa227d231/rllib/core/rl_module/default_model_config.py).

There's an additional, related issue in line 163 of primitives.py that causes encoders with layer normalization not to use it on their last layer _(since it uses the same logic as an action or value head, which shouldn't do so)_:

https://github.com/ray-project/ray/blob/master/rllib/core/models/torch/primitives.py#L163

I would solve this by applying layer normalization so long as an output layer activation is used, since it seems like that's always what you'd want to do:

```
class TorchMLP(nn.Module):
    def __init__(... ):
        super().__init__()
        assert input_dim > 0

        self.input_dim = input_dim

        hidden_activation = get_activation_fn(
            hidden_layer_activation, framework="torch"
        )
        output_activation = get_activation_fn( # Moved this up
            output_activation, framework="torch"
        )
       ...

            # Changed this section
            activation = output_activation if is_output_layer else hidden_activation
            # Insert a layer normalization in between layer's output and
            # the activation.
            if hidden_layer_use_layernorm and (activation is not None):
                # We use an epsilon of 0.001 here to mimick the Tf default behavior.
                layers.append(nn.LayerNorm(dims[i + 1], eps=0.001))
            # Add the activation function.
            if activation is not None:
                layers.append(activation())
                
        self.mlp = nn.Sequential(*layers)
...
```

I'm happy to submit a PR if this fix looks right to you.

### Versions / Dependencies

Ray 2.53.0
Python 3.12.12
Ubuntu 22.04.5 LTS

### Reproduction script

```
from ray.rllib.algorithms.ppo.torch.default_ppo_torch_rl_module import (  # noqa
    DefaultPPOTorchRLModule,
)
from ray.rllib.core.rl_module.rl_module import RLModuleSpec
import gymnasium as gym

RLModuleSpec(
    module_class=DefaultPPOTorchRLModule,
    observation_space=gym.spaces.Box(0,1,(10,)),
    action_space=gym.spaces.Box(0,1,(10,)),
    model_config={
        #
        "fcnet_hiddens": [16] * (3-1),
        "head_fcnet_hiddens": [16] * (3-1),
        "vf_share_layers": False, 
        "fcnet_use_layernorm": True,
        "fcnet_activation": 'relu',
    },
).build()

# no LayerNorm layers will be added without the patches above
```

**Comment by HassamSheikh** [comment]
```
from ray.rllib.algorithms.ppo.torch.default_ppo_torch_rl_module import DefaultPPOTorchRLModule
from ray.rllib.core.rl_module.rl_module import RLModuleSpec
import gymnasium as gym

model = RLModuleSpec(
    module_class=DefaultPPOTorchRLModule,
    observation_space=gym.spaces.Box(0,1,(10,)),
    action_space=gym.spaces.Box(0,1,(10,)),
    model_config={
        'fcnet_hiddens': [16] * (3-1),
        'head_fcnet_hiddens': [16] * (3-1),
        'vf_share_layers': False, 
        'fcnet_use_layernorm': True,
        'fcnet_activation': 'relu',
    },
).build()

# Search for LayerNorm in the model
import torch.nn as nn
layernorm_modules = []
for name, module in model.named_modules():
    if isinstance(module, nn.LayerNorm):
        layernorm_modules.append((name, module))

print('LayerNorm layers found:', len(layernorm_modules))
for name, mod in layernorm_modules:
    print(f'  {name}: {mod} (normalized_shape={mod.normalized_shape})')
print()
print('Full model structure (encoder):')
print(model.encoder.actor_encoder.net)
```

@MatthewCWeston  I updated the debugging script, it indeed doesn't include layer_norm in the neural network.
